### PR TITLE
SONAR-22581 docker image openshift hardening

### PR DESCRIPTION
**Pull request by jCOTINEAU**

### CODE REVIEW

**Review by carminevassallo — CHANGES_REQUESTED** [review]
Hi Jeremy,

great job! I was mentioning a few things in my review, could you please have a look and tell me what you think?

Furthermore, the `sq-dce-postgres` compose file does not work. I get several directories that cannot be created (I'm using images built after your changes). Could you please up this docker compose and tell me if you get similar errors?

**Comment by jCOTINEAU** [comment]
> Hi Jeremy,
> 
> great job! I was mentioning a few things in my review, could you please have a look and tell me what you think?
> 
> Furthermore, the `sq-dce-postgres` compose file does not work. I get several directories that cannot be created (I'm using images built after your changes). Could you please up this docker compose and tell me if you get similar errors?

Well it is weird, I have been testing only with this docker-compose file and the new images, let me double check this.

File: example-compose-files/sq-dce-postgres/docker-compose.yml [hunk 1]
```diff
@@ -106,6 +115,8 @@ services:
       SONAR_CLUSTER_NODE_NAME: "search-3"
     volumes:
       - search-data-3:/opt/sonarqube/data
+      - sonarqube_logs:/opt/sonarqube/logs
```
[the remark this answers is not in the document]
Comment (jCOTINEAU) [reply example-compose-files/sq-dce-postgres/docker-compose.yml:118]: indeed ultimately we should have multiple volumes, nonetheless considering it is an example only and people should rely on docker-compose logs to get them.

But yeah it does not cost much to split into three volumes

File: 10/community/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,10 @@
 FROM eclipse-temurin:17-jre-jammy
 
+LABEL io.k8s.description="SonarQube is a self-managed, automatic code review tool that systematically helps you deliver Clean Code."
```
[the remark this answers is not in the document]
Comment (jCOTINEAU) [reply 10/community/Dockerfile:3]: Those labels cannot be added dinamically they are here for sorting and filtering docker images.

I think its great to keep them as it give some more information to users and automated tools.

**Comment by jCOTINEAU** [comment]
> > Hi Jeremy,
> > great job! I was mentioning a few things in my review, could you please have a look and tell me what you think?
> > Furthermore, the `sq-dce-postgres` compose file does not work. I get several directories that cannot be created (I'm using images built after your changes). Could you please up this docker compose and tell me if you get similar errors?
> 
> Well it is weird, I have been testing only with this docker-compose file and the new images, let me double check this.

Right some issues were happening on the app nodes, missed that during the tests, fixed

File: 10/community/Dockerfile [hunk 3]
```diff
@@ -47,7 +60,10 @@ RUN set -eux; \
     chmod -R 550 ${SONARQUBE_HOME}; \
     chmod -R 770 "${SQ_DATA_DIR}" "${SQ_EXTENSIONS_DIR}" "${SQ_LOGS_DIR}" "${SQ_TEMP_DIR}"; \
     apt-get remove -y gnupg unzip; \
-    rm -rf /var/lib/apt/lists/*;
+    rm -rf /var/lib/apt/lists/*; \
+    apt-get clean;
```
[the remark this answers is not in the document]
Comment (jCOTINEAU) [reply 10/community/Dockerfile:64]: Both are not doing exactly the same thing; nonetheless, even if we are following best practices, we are not saving much space here nor reducing surface attacks.

File: example-compose-files/sq-dce-postgres/docker-compose.yml [hunk 4]
```diff
@@ -39,6 +39,7 @@ services:
       - sonarqube_extensions:/opt/sonarqube/extensions
       - sonarqube_logs:/opt/sonarqube/logs
       - sonarqube_temp:/opt/sonarqube/temp
+      - /opt/sonarqube/data
```
Comment (carminevassallo) [inline example-compose-files/sq-dce-postgres/docker-compose.yml:42]: I don't get this. Is that intended?

**Review by carminevassallo — CHANGES_REQUESTED** [review]
A few small other requests

File: example-compose-files/sq-dce-postgres/docker-compose.yml (hunk 4 above)
Comment (jCOTINEAU) [reply example-compose-files/sq-dce-postgres/docker-compose.yml:42]: Yes, in order to have one volume per replica the only solution is to use unnamed volume, which to me is completely fine as we are not binding them locally.

The other solution would have been to split the 3 replicas into 3 separate service like for the search nodes

Comment (carminevassallo) [reply example-compose-files/sq-dce-postgres/docker-compose.yml:42]: good point, thanks for clarifying!

**Review by carminevassallo — APPROVED** [review]

**Pull request merged**